### PR TITLE
Fix line number width bug and add syntax highlighting

### PR DESCRIPTION
**Pull request by SMores**
### Description of Change

Fixes the line number width bug identified in #159, and adds support for hightlight.js for syntax highlighting. ~The highlight.js bundle and default languages adds about 40kB total gzipped, but they're all marked deferred, and so shouldn't interfere with page load time.~

This runs highlight.js on the backend, so the only new asset that needs to be loaded on the frontend is the highlight.js CSS, which is a few hundred bytes.

### Motivation and Context

See #159 for line numbering context. Here's how this looks with fixed line numbers and syntax highlighting:

Screen Shot 2020-06-23 at 12 23 17 PM

### Checklist

- [x] Ran `npm run lint` and updated code style accordingly
- [x] `npm run test` passes
- [x] PR has a description and all contributors/stakeholder are noted/cc'ed
- [x] tests are updated and/or added to cover new code
- [ ] relevant documentation is changed and/or added

### CODE REVIEW

**Comment by SMores** [comment]
I don't think there are any needed documentation changes here, but happy to make them if there are. The tests passed locally but... the linter didn't, because of this line: https://github.com/nytimes/library/blob/master/server/formatter.js#L86. It looks like eslint can't parse lookbehinds? Possibly just an eslint version issue? (I'm seeing the same error in my code editor:
image

**Comment by isaacwhite** [comment]
There's an open issue for the eslint problem #141, which we've been careful to address because we don't want to break the regex by changing it. If it is an eslint version issue as you suggest, I'd be happy for us to upgrade to a newer version!

**Comment by isaacwhite** [comment]
I think you're right about the eslint version being the culprit; we're pretty far behind there. What if we merge [this PR](https://github.com/nytimes/library/pull/157) and then apply it to your branch, to get the linter to pass?

**Comment by SMores** [comment]
Looks like that did it, linting is passing locally now!

File: server/formatter.js [hunk 1]
```diff
@@ -107,7 +108,9 @@ function formatCode(html) {
     // strip interior <p> tags added by google
     content = content.replace(/<\/p><p>/g, '\n').replace(/<\/?p>/g, '')
 
-    return `<pre type="${codeType}">${formatCodeContent(content)}</pre>`
+    const formattedContent = formatCodeContent(content)
+    const highlighted = codeType ? hljs.highlight(codeType, unescape(formattedContent), true) : hljs.highlightAuto(unescape(formattedContent))
+    return `<pre><code class=${codeType}>${highlighted.value}</code></pre>`
```
Comment (isaacwhite) [inline server/formatter.js:113]: 💯  Love that we can do this in the backend.

**Comment by isaacwhite** [comment]
One other option worth exploring, too: using an npm like https://www.npmjs.com/package/striptags instead of `unescape` for this use case.

**Comment by SMores** [comment]
Ahhh interesting. I'm definitely fine with not using "auto" to try to guess the language if there's no code type available, but let me poke around and see if there are other ways to solve this. That specific case is kind of funky, honestly... it looks like a weird copy/paste; there's no reason for `git@github.com` to be treated as an email address there. Maybe we should run the highlighter on the text content of the node instead of its innerHTML, which would just ignore the Docs formatting for that case (it would be correct to do that there, but I don't know if that's true in other scenarios, necessarily)

**Pull request merged**